contracts/core/Pool.sol

```diff
@@ -230,6 +256,11 @@ contract Pool is IPool, PoolERC20 {
     uint256 amount1Out,
     address to
   ) public override lock nonZeroAddress(to) {
+    // Hmmm why is this function public actually?
+    // Also, this piece of logic is strange since now the calculation logic is in the contract already
+    // => From the amountIn, you can directly calculate the amount out, instead of the current logic
+    // where you calculate the amountOut and then call this function to once again verify the K
+
```

Comment:
That is very true haha. Original intention of swap was to allow users to to do a flash swap and allow it as long as they return back whatever is needed, but then we did not implement a variant of UniswapV2Callee in the end so...

Reply:
LOL I see. Even for flashswap right, the user can still provide the amountIn, you transfer the amountOut out, do the callback, then just simply checks enough amountIn has been transferred in. The pattern of amount0Out & amount1Out is actually appropriate only when the Router does all the logic & the pool contains minimal logic, which is not the case here